### PR TITLE
Improve error message when creating a wallet in place of one that already exists

### LINKED ISSUE #7264: Creating HD wallet at site of already-existing wallet gives cryptic error.
#🐞 Bug Report

If you have an HD wallet at location A and issue a wallet-v2 create with a wallet-dir that is also A, and if the password for the new wallet does not match the old wallet's password, you receive this:

`[2020-09-17 12:20:42] FATAL accounts-v2: Could not create a wallet: could not initialize wallet with derived keymanager: could not initialize keymanager: could not initialize derived keymanager: could not decrypt seed configuration with password: invalid checksum`

If passwords of present-wallet and what is provided to the program for wallet-v2 create are identical, then there would be no new wallet created at the location even though you get this message: 

`INFO accounts-v2: Successfully created HD wallet and saved configuration to disk. Make a new validator account with ./prysm.sh validator accounts-2 create --wallet-dir=/root/.eth2validators/prysm-wallet-v2`

This doesn't exactly fit expected behavior either. 

### Has this worked before in a previous version?
Not for sure.

## 🔬 Minimal Reproduction
See above

## 🌍  Your Environment

**Operating System:**

I reproduced with current master on Ubuntu VM running docker.

### PR DESCRIPTION
**Pull request by dv8silencer**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**
Currently, if you invoke wallet-v2 create with a wallet directory targeting a location that already has a wallet, it simply tries to use the provided [new wallet] password to unlock the old wallet without actually overwriting the old wallet.  A silent overwrite is not desirable either.  If the provided password mismatches with the old wallet's password, you get an error message:

`[2020-09-17 12:20:42] FATAL accounts-v2: Could not create a wallet: could not initialize wallet with derived keymanager: could not initialize keymanager: could not initialize derived keymanager: could not decrypt seed configuration with password: invalid checksum`

If the password provided for the new wallet matches the one needed to unlock the one already there, you get this:

`INFO accounts-v2: Successfully created HD wallet and saved configuration to disk. Make a new validator account with ./prysm.sh validator accounts-2 create --wallet-dir=/root/.eth2validators/prysm-wallet-v2`

Which is not accurate.  Also, in this case the keymanageropts.json file ends up overwritten in the code flow by a default one.

This PR makes it so that if wallet-v2 create is invoked to create the same **type** of wallet already present, it will stop and output an error message: 

> a wallet of this type already exists at this location. Please input an alternative location for the new wallet or remove the current wallet

**Which issues(s) does this PR fix?**

Fixes #7264 

**Other notes for review**
Has a regression test.

### CODE REVIEW

File: validator/accounts/v2/wallet_create_test.go [hunk 1]
```diff
@@ -217,6 +217,38 @@ func TestCreateWallet_Derived(t *testing.T) {
 	assert.DeepEqual(t, derived.DefaultKeymanagerOpts(), cfg)
 }
 
+// TestCreateWallet_IfExistingWallet checks for expected error if trying to create a wallet where they is already one.
+func TestCreateWallet_IfExistingWallet(t *testing.T) {
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:221]: `TestCreateWallet_WalletAlreadyExists` would be a name that is more in line with the rest of the tests in the project.

Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:221]:
```suggestion
func TestCreateWallet_WalletAlreadyExists(t *testing.T) {
```

File: validator/accounts/v2/wallet_create_test.go [hunk 2]
```diff
@@ -217,6 +217,38 @@ func TestCreateWallet_Derived(t *testing.T) {
 	assert.DeepEqual(t, derived.DefaultKeymanagerOpts(), cfg)
 }
 
+// TestCreateWallet_IfExistingWallet checks for expected error if trying to create a wallet where they is already one.
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:220]:
```suggestion
// TestCreateWallet_WalletAlreadyExists checks for expected error if trying to create a wallet when there is one already.
```

File: validator/accounts/v2/wallet_create.go [hunk 3]
```diff
@@ -36,6 +38,19 @@ func CreateAndSaveWalletCli(cliCtx *cli.Context) (*wallet.Wallet, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	extractedWalletDir := createWalletConfig.WalletCfg.WalletDir
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create.go:42]:
```suggestion
	walletDir := createWalletConfig.WalletCfg.WalletDir
```

File: validator/accounts/v2/wallet_create.go [hunk 4]
```diff
@@ -36,6 +38,19 @@ func CreateAndSaveWalletCli(cliCtx *cli.Context) (*wallet.Wallet, error) {
...
+	extractedKeymanagerKind := createWalletConfig.WalletCfg.KeymanagerKind
+	accountsPath := filepath.Join(extractedWalletDir, extractedKeymanagerKind.String())
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create.go:44]:
```suggestion
	walletDir := createWalletConfig.WalletCfg.WalletDir
	keymanagerKind := createWalletConfig.WalletCfg.KeymanagerKind
	accountsPath := filepath.Join(walletDir, keymanagerKind.String())
```

File: validator/accounts/v2/wallet_create_test.go [hunk 5]
```diff
@@ -217,6 +217,38 @@ func TestCreateWallet_Derived(t *testing.T) {
...
+		passwordsDir:       passwordsDir,
+		walletPasswordFile: passwordFile,
+		keymanagerKind:     v2keymanager.Derived,
+	})
+
+	// We attempt to create the wallet.
+	_, err := CreateAndSaveWalletCli(cliCtx)
+	require.NoError(t, err)
+
+	// We attempt to create another wallet of the same type at the same location.  We expect an error.
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:234]:
```suggestion
	// We attempt to create another wallet of the same type at the same location. We expect an error.
```

File: validator/accounts/v2/wallet_create_test.go [hunk 6]
```diff
@@ -217,6 +217,38 @@ func TestCreateWallet_Derived(t *testing.T) {
...
+		"Creating a wallet of the same type at an existing wallet location should return error")
+
+	cliCtx = setupWalletCtx(t, &testWalletConfig{
+		walletDir:          walletDir,
+		passwordsDir:       passwordsDir,
+		walletPasswordFile: passwordFile,
+		keymanagerKind:     v2keymanager.Direct,
+	})
+
+	// We attempt to create another wallet of different type at the same location.  We don't expect an error.
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:246]:
```suggestion
	// We attempt to create another wallet of different type at the same location. We don't expect an error.
```

File: validator/accounts/v2/wallet_create_test.go [hunk 7]
```diff
@@ -217,6 +217,38 @@ func TestCreateWallet_Derived(t *testing.T) {
...
+	_, err = CreateAndSaveWalletCli(cliCtx)
+	require.Equal(t, nil, err,
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:248]: You can use `require.NoError`

File: validator/accounts/v2/wallet_create_test.go [hunk 8]
```diff
@@ -217,6 +217,38 @@ func TestCreateWallet_Derived(t *testing.T) {
...
+		keymanagerKind:     v2keymanager.Derived,
+	})
+
+	// We attempt to create the wallet.
+	_, err := CreateAndSaveWalletCli(cliCtx)
+	require.NoError(t, err)
+
+	// We attempt to create another wallet of the same type at the same location.  We expect an error.
+	_, err = CreateAndSaveWalletCli(cliCtx)
+	require.NotEqual(t, nil, err,
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:236]: Please use `require.ErrorContains` and verify the error message.

**Review by rkapka — CHANGES_REQUESTED** [review]

File: validator/accounts/v2/wallet_create_test.go (hunk 8 above)
Comment (rkapka) [reply validator/accounts/v2/wallet_create_test.go:236]: I think the test in its current will pass even with the old code.

File: validator/accounts/v2/wallet_create_test.go (hunk 7 above)
Comment (dv8silencer) [reply validator/accounts/v2/wallet_create_test.go]: Addressed.  Thanks

File: validator/accounts/v2/wallet_create_test.go (hunk 8 above)
Comment (dv8silencer) [reply validator/accounts/v2/wallet_create_test.go]: Hey I made a branch from current main that only has the test:  https://github.com/dv8silencer/prysm/tree/dv8-PR-7268-Regression  This is the diff vs current master:  https://github.com/prysmaticlabs/prysm/compare/master...dv8silencer:dv8-PR-7268-Regression?expand=1

When running:
```
root@112af41569a9:~/prysm/validator/accounts/v2# go test ./... --tags=blst_enabled -v | grep TestCreateWallet_WalletAlreadyExists
=== RUN   TestCreateWallet_WalletAlreadyExists
--- FAIL: TestCreateWallet_WalletAlreadyExists (0.01s)
```

When running PR branch:
```
root@112af41569a9:~/prysm/validator/accounts/v2# go test ./... --tags=blst_enabled -v | grep TestCreateWallet_WalletAlreadyExists
=== RUN   TestCreateWallet_WalletAlreadyExists
--- PASS: TestCreateWallet_WalletAlreadyExists (0.01s)
```

File: validator/accounts/v2/wallet_create_test.go [hunk 9]
```diff
@@ -233,7 +233,8 @@ func TestCreateWallet_WalletAlreadyExists(t *testing.T) {
 
 	// We attempt to create another wallet of the same type at the same location. We expect an error.
 	_, err = CreateAndSaveWalletCli(cliCtx)
-	require.NotEqual(t, nil, err,
+	require.ErrorContains(t, "a wallet of this type already exists at this location. Please input an"+
+		" alternative location for the new wallet or remove the current wallet", err,
 		"Creating a wallet of the same type at an existing wallet location should return error")
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:238]: This argument can be omitted as it does not add much value.

File: validator/accounts/v2/wallet_create_test.go [hunk 10]
```diff
@@ -245,8 +246,8 @@ func TestCreateWallet_WalletAlreadyExists(t *testing.T) {
 
 	// We attempt to create another wallet of different type at the same location. We don't expect an error.
 	_, err = CreateAndSaveWalletCli(cliCtx)
-	require.Equal(t, nil, err,
-		"Creating a wallet of a different type at an existing wallet location should not return a error")
+	require.NoError(t, err, "Creating a wallet of a different type at an existing wallet location"+
```
Comment (rkapka) [inline validator/accounts/v2/wallet_create_test.go:249]: The last argument can be omitted as it does not add much value.

File: validator/accounts/v2/wallet_create_test.go (hunk 9 above)
Comment (Bit24888) [reply validator/accounts/v2/wallet_create_test.go:238]: kQBWkNKqzCAwA9vjMwRmg7aY75Rf8lByPA9zKXoqGkHi8SM7

**Review by rkapka — APPROVED** [review]

**Pull request merged**